### PR TITLE
Add VectorUtils and SDF

### PR DESCRIPTION
**Pull request by 4leph0mega**
It adds functionality on 3D vectors more easily

### CODE REVIEW

File: extensions/AlephOmega/VectorUtils.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
-
+//I HATE ESLING I HATE ESLIN I HATE ESLINT I HAVE ESLINT I HATE ESLING I HATE SLINT I HAVE TO REWRITE THE ENTIRE PROGRAM
```
Comment (GarboMuffin) [inline extensions/AlephOmega/VectorUtils.js]: `npm run fix`

Comment (4leph0mega) [reply extensions/AlephOmega/VectorUtils.js]: pl

Comment (4leph0mega) [reply extensions/AlephOmega/VectorUtils.js]: ok*

Comment (4leph0mega) [reply extensions/AlephOmega/VectorUtils.js:1]: done

Comment (4leph0mega) [reply extensions/AlephOmega/VectorUtils.js:1]: done

Comment (Skyhigh173) [reply extensions/AlephOmega/VectorUtils.js]: > ok*

do you know you can edit your messages

Comment (4leph0mega) [reply extensions/AlephOmega/VectorUtils.js]: i actually didn't know

File: extensions/AlephOmega/VectorUtils.js [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+class VectorExtension {
+  getInfo() {
+    return {
+      id: 'vector',
```
Comment (GarboMuffin) [inline extensions/AlephOmega/VectorUtils.js:11]: extension ids must be somehow based on your username, eg `alpephomegavector`

File: extensions/AlephOmega/SDFs.js [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+    }
+  
+    class SDF {
+      getInfo() {
+        return {
+          color1: "#ad0090",
+          color2: "#6e145f",
+          color3: "#3b0932",
+          id: 'SDF',
+          name: 'SDFs',
```
Comment (GarboMuffin) [inline extensions/AlephOmega/SDFs.js:59]: same here, eg. `alpephomegasdf`

**Comment by LukeManiaStudioss** [comment]
The blocks don't have any error handling, so every single block (except the 2 blocks in the image) stalls if you click it in the palette.
image
image

**Comment by GarboMuffin** [comment]
thank you for your patience

if you still would like this to be added, I would like to see:

 - improved error handling for invalid JSON (I bet you can find a more compact way yo represent the vectors too)
 - resolve all ESLint errors in the checks at the bottom of these comments (warnings are okay; I can resolve those myself later) 
 - may help to clarify what an SDF is
 - the default values in the blocks should ideally demonstrate the block somehow, at least not throw errors

**Comment by 4leph0mega** [comment]
> 

thanks for the feedback, i dont really have much free time to work on it currently but i will try to do it

**Comment by GarboMuffin** [comment]
Sorry for the delay

I am closing this in favor of https://github.com/TurboWarp/extensions/pull/779 which seems to have the same vector utilities

If you are still interested in getting the SDF extension in then please let me know

**Comment by GarboMuffin** [comment]
Actually, I'll leave this open, but only for the SDF extension

**Comment by GarboMuffin** [comment]
sorry, SDFs has some issues

if you're interested in continuing with this, please let me know, but i do need to clear things out of the pull request queue if they are abandoned

**Pull request closed without merging**